src/thunderbird_accounts/mail/clients.py

```diff
@@ -513,84 +424,129 @@ def get_dns_records(self, domain_name: str) -> list[dict]:
         return data.get('data')
 
     def verify_domain(self, domain_name: str):
```

Comment:
For the view we might want to look into rate limiting and moving this function to task in the near future